sw/device/tests/kmac_entropy_test.c

```diff
@@ -0,0 +1,232 @@
+               0x5D3030CC},
+    .digest_len = 16,
+    .digest_len_is_fixed = false,
+};
+
+bool test_main(void) {
+  LOG_INFO("Running KMAC ENTROPY test...");
+
+  // Initialize KMAC HWIP
+  dif_kmac_t kmac;
+  CHECK_DIF_OK(
+      dif_kmac_init(mmio_region_from_addr(TOP_EARLGREY_KMAC_BASE_ADDR), &kmac));
+
+  // Configure KMAC to use EDN entropy
+  dif_kmac_config_t config = {
+      .entropy_mode = kDifKmacEntropyModeEdn,
+      .entropy_fast_process = false,
+      // In the last iteration, we expect to see that hash_ctr rewinds to 0
+      .entropy_hash_threshold = kKmacTimeoutTestCount,
+      .message_big_endian = false,
+      .output_big_endian = false,
+      .sideload = false,
+      .msg_mask = false,
+  };
+
+  // Handle string encoding
+  dif_kmac_customization_string_t str_buffer;
+  CHECK_DIF_OK(dif_kmac_customization_string_init(
+      kmac_test_vector.customization_string,
+      kmac_test_vector.customization_string_len, &str_buffer));
+
+  // When customization_string is empty, use NULL to activate empty str path
+  dif_kmac_customization_string_t *s_buff =
+      kmac_test_vector.customization_string_len == 0 ? NULL : &str_buffer;
+
+  size_t digest_l =
+      kmac_test_vector.digest_len_is_fixed ? kmac_test_vector.digest_len : 0;
+
+  for (size_t i = 0; i < kKmacTimeoutTestCount; i++) {
+    LOG_INFO("loop_ctr = %d", i);
+
+    kmac_edn_timeout_t timeout_test;
+    timeout_test = kTestTimeoutVals[i];
+
+    config.entropy_prescaler = timeout_test.prescaler;
+    config.entropy_wait_timer = timeout_test.wait_timer;
+
+    CHECK_DIF_OK(dif_kmac_configure(&kmac, config));
+
+    dif_kmac_operation_state_t kmac_operation_state;
+    CHECK_DIF_OK(dif_kmac_mode_kmac_start(&kmac, &kmac_operation_state,
+                                          kmac_test_vector.mode, digest_l,
+                                          &kmac_test_vector.key, s_buff));
+
+    CHECK_DIF_OK(dif_kmac_absorb(&kmac, &kmac_operation_state,
+                                 kmac_test_vector.message,
+                                 kmac_test_vector.message_len, NULL));
+
+    CHECK(kKmacDigestLenMax >= kmac_test_vector.digest_len);
+
+    // This is where timeout might happen, so we handle dif return manually
+    uint32_t out[kKmacDigestLenMax];
+    size_t *processed_len = NULL;
+    dif_result_t res =
+        dif_kmac_squeeze(&kmac, &kmac_operation_state, out,
+                         kmac_test_vector.digest_len, processed_len);
+
+    // It is OK to get kDifError at this point because of possible timeout
+    CHECK(res == kDifOk || res == kDifError);
+
+    // Check if there is a new error
+    bool irq_err_pending;
+    CHECK_DIF_OK(
+        dif_kmac_irq_is_pending(&kmac, kDifKmacIrqKmacErr, &irq_err_pending));
+    if (irq_err_pending) {
+      dif_kmac_error_t err_status;
+      CHECK_DIF_OK(dif_kmac_get_error(&kmac, &err_status));
+      CHECK(err_status == kDifErrorNone ||
```

Comment:
Two questions: 1)  Can `err_status` be `kDifErrorNone`? 2) Can there be more than one error pending? If so, should this be a loop?